redis/ocsp.py

```diff
@@ -11,8 +11,50 @@
 from redis.exceptions import AuthorizationError, ConnectionError
 
 
+def _check_certificate(certificate):
+    """A wrapper the return the validity of a known ocsp certificate"""
+
+    ocsp_response = ocsp.load_der_ocsp_response(certificate)
```

Comment:
The response contains a `thisUpdate` timestamp, the verifier should check that it is valid (i.e. in the past)
the response **may** contain a `nextUpdate` timestamp as well, stating until when is this stapled response valid. if it exists in the response, you should check that the response is still valid

Reply:
_ocsp_response.this_update_ being in the part is fair - the check is ready. If I read [RFC 5019](https://www.rfc-editor.org/rfc/rfc5019#page-7) correctly, a valid responder must produce a response, to allow for caching. This means the cert check will now raise a connection error.  Please correct me if I'm wrong.

Reply:
nope. `this_update` should simply be in the past, meaning that the current time should not be before `this_update`
even cached responses should be valid between `this_update` and `next_update`, which we should verify that is true (unless `next_update` is not there, then only check that we're after `this_update` timestamp)

Reply:
Perhaps I wasn't being clear. In [section 2.2.4](https://www.rfc-editor.org/rfc/rfc5019#section-2.2.4) the RFC states that _Responders MUST always include this value to aid in response caching_, WDYT?

Reply:
This appears to be superceded by [RFC 6960#4.2.2.1](https://www.rfc-editor.org/rfc/rfc6960#section-4.2.2.1). But given the language it's unclear. Erring on the side of not requiring a nextUpdate.